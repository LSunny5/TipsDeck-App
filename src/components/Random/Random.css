.RandomContent {
    margin: 0;
    padding-bottom: 5rem;
    padding-top: 2.3rem;
}

.RandomContent > h1 {
    font-size: 2em;
    padding: 3%;
    font-family: 'Della Respira', serif;
}

  /*for ipads and larger screen sizes*/
  @media only screen and (min-width: 600px) {
    .RandomContent {
      padding-bottom: 5rem;
      min-height: 97vh;
      padding-top: 5.8rem;
    }
    
    .RandomContent > h1 {
          padding: 1%;
      }
    }

      /*for desktops*/
  @media only screen and (min-width: 1024px) {
    .RandomContent{
      min-height: 92vh;
      padding-top: 1rem;
      padding-bottom: 4rem;
    }
  }