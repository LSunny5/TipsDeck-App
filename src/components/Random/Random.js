import React from 'react';
import './Random.css';
import TipsDeckContext from '../../TipsDeckContext';
import { shuffleArray} from '../../TipsDeckHelpers';
import { NavLink } from 'react-router-dom';

class Random extends React.Component {
    static contextType = TipsDeckContext;

    render() {
        const { tips = [] } = this.context;
        let temp = [...tips];
        temp = shuffleArray(temp);

        
        /* let tempId;
        let findCat = findTip(tips, tempId);
        let a = getCategoryName(categories, findCat.category)
        let catName = a.name; */

        return (
            <section className="RandomContent">
                <h1>Random</h1>

                <div className="TipsBox">
                    {(temp.length > 0) ? (
                        temp.map(tip =>
                            <NavLink
                                key={tip.id}
                                to={`/Category/${tip.category}/${tip.id}`}
                                className="TipLink">
                                <h2 className="tName">{tip.name}</h2>
                                <p className="tipDesc">{tip.description}</p>
                            </NavLink>
                        )) : (
                            <div className="noTips">
                                Sorry there are no tips at this time.
                            </div>)}
                </div>
            </section>
        );
    }
}

export default Random;
