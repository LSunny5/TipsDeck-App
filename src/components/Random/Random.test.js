import React from 'react';
import ReactDOM from 'react-dom';
import Random from './Random';
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import { BrowserRouter } from 'react-router-dom';

describe(`Random component`, () => {
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <Random />
            </BrowserRouter>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the Random component', () => {
        const wrapper = shallow(<Random />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
});
