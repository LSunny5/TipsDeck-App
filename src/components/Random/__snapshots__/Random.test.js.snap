// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Random component renders the Random component 1`] = `
<section
  className="RandomContent"
>
  <h1>
    Random
  </h1>
  <div
    className="TipsBox"
  >
    <div
      className="noTips"
    >
      Sorry there are no tips at this time.
    </div>
  </div>
</section>
`;
